config/dependency.go

```diff
@@ -418,6 +429,59 @@ func canGetRemoteState(remoteState *remote.RemoteState) bool {
 	return remoteState != nil && !remoteState.DisableDependencyOptimization
 }
 
+// terragruntAlreadyInit returns true if it detects that the module specified by the given terragrunt configuration is
+// already initialized with the terraform source. This will also return the working directory where you can run
+// terraform.
+func terragruntAlreadyInit(terragruntOptions *options.TerragruntOptions, configPath string) (bool, string, error) {
+	// We need to first determine the working directory where the terraform source should be located. This is dependent
+	// on the source field of the terraform block in the config.
+	terraformBlockTGConfig, err := PartialParseConfigFile(configPath, terragruntOptions, nil, []PartialDecodeSectionType{TerraformSource})
+	if err != nil {
+		return false, "", err
+	}
+	var workingDir string
+	sourceUrl := GetTerraformSourceUrl(terragruntOptions, terraformBlockTGConfig)
+	if sourceUrl == "" || sourceUrl == "." {
```

Comment:
When would it be set to "."?

Reply:
This is more defensive coding for certain user behavior, where one might decide that it is better to be explicit instead of implicit.

Reply:
Roger